_datafiles/world/default/mobs/frostfang/scripts/2-guard.js

```diff
@@ -1,15 +1,29 @@
 
+var PathTargets = [
+    [1, 59, 263, 'home'],          // town square => east gate => east residential district => home
+    [781, 35, 1, 'home'],          // west residential district => west gate => town square => home
+    [166, 63, 76, 62, 61, 'home'], // bank => steelwhisper armory => Hacking Hut => Icy Emporium => Inn => home
+    [16, 'home'],                  // south end of beggars lane => home
+    [44, 42, 'home'],              // east castle wing => west castle wing => home
+];
+
 function onIdle(mob, room) {
     
-    var random = Math.floor(Math.random() * 8);
+    var random = Math.floor(Math.random() * 10);
```

Comment:
Magic number?

Reply:
Just a random 1 in 10 to pick an action. The 1 in 8 (12%) is less intuitive.